File: search-insert-position/SearchInsertPosition.swift
class SearchInsertPosition {
    func searchInsert(_ nums: [Int], _ target: Int) -> Int {
        var left = 0
        var right = nums.count - 1
        while left <= right {
            if nums[left] == target {
                return left
            } else if nums[left] < target {
                left += 1
            } else {
                right -= 1
            } 
        }
        return left
    }
}

File: search-insert-position/SearchInsertPosition.java

public class SearchInsertPosition {
    public int searchInsert(int[] nums, int target) {
        int left = 0;
        int right = nums.length - 1;
        while(left <= right) {
            if(nums[left] == target) {
                return left;
            } else if(nums[left] < target) {
                left++;
            } else {
                right--;
            }
        }
        return left;
    }
}
